WalletWasabi/Wallets/WalletManager.cs

```diff
@@ -46,14 +59,38 @@ public WalletService GetFirstOrDefaultWallet()
 			}
 		}
 
-		public async Task AddAndStartAsync(WalletService walletService, CancellationToken cancel)
+		public async Task<WalletService> AddWalletServiceAsync(KeyManager keyManager)
 		{
+			WalletService walletService;
+
+			while (!WalletCreationEnabled)
```

Comment:
We are using this pattern in many places and I think it is not good because it is expensive in CPU terms. It would be better to switch to a signaling mechanism like the one use with `WaitHandle` (more spacifically `ManuaResetWaitHandle(Slim?)`). Take a look at: https://thomaslevesque.com/2015/06/04/async-and-cancellation-support-for-wait-handles/

this could be used as follow:
```c#
await WalletCreationWaiter.WaitOneAsync(StoppongCts.Token);
```

**This just a comment to share my thoughts and no a change request**

Reply:
I thought it was obvious. We are moving towards that for half a year now. `TaskCompletionSource` btw, so to stay inside the async await task paradigm. I think this was just copypasted (moved) code that's why it's like this.

Reply:
Btw the old pattern isn't problematic for the CPU. Also let's avoid premature optimizations unless there' a proven bottleneck and concentrate on readability instead, premature optimizations are biting back more and more often.

Reply:
I mentioned the CPU because the the readability improvement of using a one-line mechanism is self-evident. You're right, `TaskCompletionSource` is the way to go, however, I am not requesting any change here.